Added support for Ada Language Server

thindil opened the pull request.

w0rp commented on ale_linters/ada/adalsp.vim:
```diff
@@ -0,0 +1,16 @@
+endfunction
+
+call ale#linter#Define('ada', {
+\   'name': 'adalsp',
```
Let's name this `adals`, and name the variable aboev `ada_adals_executable`. That will match some existing linter names like `dls` for D, and how the options are named for that.

timlag1305 commented on ale_linters/ada/adals.vim:
```diff
@@ -0,0 +1,16 @@
+" Author: Bartek Jasicki http://github.com/thindil
+" Description: Support for Ada Language Server
+
+call ale#Set('ada_adals_executable', 'ada_language_server')
```
I was looking into adding the Ada language server a while back and totally forgot about it until now. In any case, I added a couple more variables that the user could set:
`ada_ls_project` and `ada_ls_encoding` which correspond to "ada.projectFile" and "ada.defaultCharset" as listed here https://github.com/AdaCore/ada_language_server/#how-to-use-the-vscode-extension

I could give you the diff if you're interested.

thindil replied:
That will be awesome :) Sure, please send me a diff, I will try add it as soon as possible.

timlag1305 replied:
```suggestion
call ale#Set('ada_adals_executable', 'ada_language_server')
call ale#Set('ada_adals_project, 'default.gpr')
call ale#Set('ada_adals_encoding', 'utf-8')

function! ale_linters#ada#adals#GetInitializationOptions(buffer) abort
    return {
    \   'ada.projectFile': ale#Var(a:buffer, 'ada_adals_project'),
    \   'ada.defaultCharset': ale#Var(a:buffer, 'ada_adals_encoding')
    \}
endfunction
```

timlag1305 commented on ale_linters/ada/adals.vim:
```diff
@@ -0,0 +1,16 @@
+\   'executable': {b -> ale#Var(b, 'ada_adals_executable')},
+\   'command': '%e',
+\   'project_root': function('ale_linters#ada#adals#GetRootDirectory'),
+\})
```
```suggestion
\   'initialization_options': function('ale_linters#ada#adals#GetInitializationOptions')
\})
```

timlag1305 commented:
I added inline diffs. I can also give you a patch file later tonight @thindil if that's easier.

thindil commented:
> I added inline diffs. I can also give you a patch file later tonight @thindil if that's easier.

@timlag1305 thank you very much. I just added them to my fork and I will test them soon. Should be everything ok.

kalkin commented:
Bump?

thindil commented:
@kalkin All requested changes are done thus now we just wait :)

timlag1305 commented:
Yeah unfortunately the original maintainer doesn't really have the time to handle all these PRs. They tried to get others to step up, but I'm not sure how effective that was. I'm sure the project would be receptive if someone was willing to step up. I think in the mean time, if I wanted a specific PR in my version of ale, I would fork the project and roll the PRs into my fork.

thindil commented:
@timlag1305 true, plus as far I know, my fork should be synced with main repo, thus you can use it instead of official and pokes me about syncing it :)

hsanson commented:
ALE migrated from Travis to Github Actions (see #3548) so it is necessary to rebase this PR from latest master branch to trigger the required checks.

```
# Add dense-analysis as remote upstream on your local fork. Not 
# needed if already added.
git remote add upstream https://github.com/dense-analysis/ale.git

# Sync your local master with upstream master
git checkout master
git fetch upstream master
git merge upstream/master

# Rebase the PR branch with master
git checkout my-branch-name
git rebase master   # Fix any conflicts you may have

# Force push the updated PR branch to origin to trigger the checks
git push -f origin my-branch-name
```

thindil commented:
@hsanson Thank you for pointing me there. The local fork was rebased as requested. Also, requested changes were done (a long time ago). If something other is needed to do here, please tell :smile:

hsanson reviewed (approved):
Thanks for the patience. Looks good to me.
